src/GafferImage/ImageWriter.cpp

```diff
@@ -869,6 +935,12 @@ IECore::MurmurHash ImageWriter::hash( const Context *context ) const
 void ImageWriter::execute() const
 {
 	// Create an OIIO::ImageOutput
+	Context * c = const_cast<Context *>( Context::current() );
+	c->set( "gaffer:colorspace", defaultColorSpace()( currentFileFormat() ) );
+
+
+	Expression *e = const_cast<Expression *>( expression() );
```

Comment:
Alright, sounds good!
When we discussed this topic initially with Daniel and Ivan that was one of the suggestion indeed.
Could this ComputeNode even works without the context at all, based on input plug use the registered function to compute a colorspace name and set an outPlug with that value?

Another thing I'm wondering is when you say remove the dependency on Python, you mean only the execution of the expression because correct me if I'm wrong but computing the colorSpace name still goes through python but I assume this is cached? Does the Expression using the Context also cache efficiently?